src/main/java/trivia/GameBetter.java

```diff
@@ -49,29 +47,32 @@ public int howManyPlayers() {
   }
 
   public void roll(int roll) {
-    System.out.println(players.get(currentPlayer) + " is the current player");
+    System.out.println(currentPlayer.getName() + " is the current player");
     System.out.println("They have rolled a " + roll);
 
-    if (!inPenaltyBox[currentPlayer]) {
+    if (currentPlayer.isNotInPenaltyBox()) {
```

Comment:
I thought about this, and my issue was that I could make a similar argument for the player's position and the board object. Is it the responsibility of the board to know where each player is? If we follow this approach, we end up with a player that can be immutable, i.e. only knows about the name of the player. Maybe this is a good thing, but it's not clear to me.

Reply:
There are two ends of the spectrum:
#1 Players who knows it all
#2 Immutable Player knowing nothing but the name

Option #1 clearly violates SRP because it gives Player many reasons to change. In that case, #2 will lead to more stable & maintainable code. But let's consider option #3.

#3 Hybrid

For me there is a subtle difference between board & jail concepts in the game.  
When you look at the board (in reality) it clearly has the knowledge of the number of the fields and fields meaning (question category). Board can clearly exist independently of its players and it has enough information to model. Game designer can change the number of the fields on the board or the meaning of the fields -> it will be reflected in the Board object.

Jail is all about players. It might not even be a palpable thing in the game, it can only be a mental construct. It consists of a players in it + the rules for getting out of it. It is reasonable to assume that Jail can answer questions like: Is Player X in the Jail, or Can Player X get out of a Jail, or Give me all the Players who are in a Jail at the moment. If Game designer changes the rules around the jailing --> there is one place in code to change. 

All in all, it is clear that Jail deserves an object on its own, and it is clear that Board deserves object on its own. Grey zone is if Board should capture the players as well.

My advise is to introduce Jail object first, see what opportunities it gives you to make code cleaner, AND in the end do an experiment with the Board+Players.